cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/runtimejob/DataprocRuntimeJobManager.java

```diff
@@ -638,23 +667,38 @@ private SubmitJobRequest getSubmitJobRequest(RuntimeJobInfo runtimeJobInfo, List
       }
     }
 
+    Job.Builder dataprocJobBuilder = Job.newBuilder()
+      // use program run uuid as hadoop job id on dataproc
+      .setReference(JobReference.newBuilder().setJobId(getJobId(runInfo)))
+      // place the job on provisioned cluster
+      .setPlacement(JobPlacement.newBuilder().setClusterName(clusterName).build())
+      // add same labels as provisioned cluster
+      .putAllLabels(labels)
+      // Job label values must match the pattern '[\p{Ll}\p{Lo}\p{N}_-]{0,63}'
+      // Since program name and type are class names they should follow that pattern once we remove all
+      // capitals
+      .putLabels(LABEL_CDAP_PROGRAM, runInfo.getProgram().toLowerCase())
+      .putLabels(LABEL_CDAP_PROGRAM_TYPE, runInfo.getProgramType().toLowerCase())
+      .setHadoopJob(hadoopJobBuilder.build());
+
+    GetClusterRequest getClusterRequest = GetClusterRequest.newBuilder()
+      .setClusterName(clusterName).setProjectId(projectId).setRegion(region).build();
+
+    // if the dataproc cluster has driver pools enabled add the driver scheduling config
+    if (getClusterControllerClient().getCluster(getClusterRequest).getConfig().getAuxiliaryNodeGroupsCount() > 0) {
+      dataprocJobBuilder.setDriverSchedulingConfig(
+        DriverSchedulingConfig.newBuilder()
+          .setMemoryMb(Integer.parseInt(provisionerContext.getProperties().getOrDefault(DataprocUtils.DRIVER_MEMORY_MB,
+                                                                           DataprocUtils.DRIVER_MEMORY_MB_DEFAULT)))
+          .setVcores(Integer.parseInt(properties.getOrDefault(DataprocUtils.DRIVER_VCORES,
```

Comment:
I believe you should use `provisionerProperties` here and above

Reply:
ack.